Silicon/CommonSocPkg/Library/TccLib/TccLib.c

```diff
@@ -0,0 +1,86 @@
+/** @file
+
+  Copyright (c) 2020 Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <PiPei.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/HobLib.h>
+#include <Library/BootloaderCommonLib.h>
+#include <Library/BootloaderCoreLib.h>
+#include <Library/TccLib.h>
+#include "TccRtctHob.h"
+
+
+// ACPI Definations for RTCT Table
+#define EFI_ACPI_RTCT_TABLE_REVISION      0x1
+#define EFI_ACPI_RTCT_OEM_ID              "INTEL "  // OEMID 6 bytes long
+#define EFI_ACPI_RTCT_OEM_TABLE_ID        SIGNATURE_64('S','B','L',' ',' ',' ',' ',' ')
+#define EFI_ACPI_RTCT_OEM_REVISION        0x00000005
+#define EFI_ACPI_RTCT_CREATOR_ID          SIGNATURE_32('I','N','T','L')
+#define EFI_ACPI_RTCT_CREATOR_REVISION    0x0100000D
+
+
+/**
+  Update RTCT  (Platform Tuning Configuration Table) ACPI Table
+
+  @param[in] RtctTable          PTCT ACPI Table entry
+
+  @retval EFI_SUCCESS           The RTCT table is updated.
+  @retval EFI_NOT_FOUND         Loader Global Pointer or TccHob not found
+  @retval EFI_INVALID_PARAMETER RTCT table is NULL
+**/
+EFI_STATUS
+EFIAPI
+UpdateAcpiRtctTable (
+  IN EFI_ACPI_DESCRIPTION_HEADER *RtctTable
+  )
+{
+  LOADER_GLOBAL_DATA            *LdrGlobal;
+  EFI_HOB_GUID_TYPE             *GuidHob;
+  EFI_ACPI_DESCRIPTION_HEADER   *HobRtctTable;
+
+  if (RtctTable == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  LdrGlobal   = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer();
+  if (LdrGlobal == NULL) {
+
+    return EFI_NOT_FOUND;
+  }
+
+  GuidHob = GetNextGuidHob (&gTccRtctHobGuid, LdrGlobal->FspHobList);

```

Comment:
Use GetFspHobListPtr() to get FSP hob list.

Reply:
OK. let me update it.